sum.go

```diff
@@ -116,8 +119,13 @@ func sumBlake2b(size uint8, data []byte) []byte {
 	return hasher.Sum(nil)[:]
 }
 
-func sumID(data []byte) []byte {
-	return data
+func sumID(data []byte, length int) ([]byte, error) {
+	if length != len(data) {
```

Comment:
this should be `if length >= 0 && length != len(data)`.

Reply:
You're right, thanks for catching this.